dipy/reconst/mcsd.py

```diff
@@ -0,0 +1,294 @@
+import numpy as np
+from dipy.core import geometry as geo
+from dipy.data import default_sphere
+from dipy.reconst import shm
+from dipy.reconst.multi_voxel import multi_voxel_fit
+
+from dipy.utils.optpkg import optional_package
+cvx, have_cvxpy, _ = optional_package("cvxpy")
+
+SH_CONST = .5 / np.sqrt(np.pi)
+
+
+def multi_tissue_basis(gtab, sh_order, iso_comp):
+    """
+    Builds a basis for multi-shell CSD model.
```

Comment:
Documentation for this function is still not complete.

Reply:
> Almost there. Just need to document these.

Sorry for that.. On it!